### PR TITLE
feat(Icon): add future Icon

### PR DESCRIPTION
**Pull request by HeartSquared**
## What and Why

Create a new Icon component which allows consumers to allow access to all icons available in [Material Symbols](https://fonts.google.com/icons).

### CODE REVIEW

File: packages/components/rollup.config.mjs [hunk 1]
```diff
@@ -9,6 +9,7 @@ export default rollupConfig({
     actionsV3: "./src/__actions__/v3.ts",
     containersV1: "./src/__containers__/v1.ts",
     containersV2: "./src/__containers__/v2.ts",
+    illustrationsV3: "./src/__illustrations__/v3.ts",
```
Comment (gyfchong) [inline packages/components/rollup.config.mjs:12]: You'll need `illustrationsV1` and `illustrationsV2`

File: packages/components/src/__illustrations__/Icon/v3/Icon.module.css [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.filled {
+  --icon-fill: 1;
+}
+
+.small {
+  --icon-optical-size: 20;
+
+  font-size: 16px;
```
Comment (gyfchong) [inline packages/components/src/__illustrations__/Icon/v3/Icon.module.css:18]: Make this a variable too...?

File: packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+### Filled
+
+<Canvas of={IconStories.Filled} />
+
+### Sizes
+
+_Default: `medium`_
+
+The `size` prop provides 3 default sizes (`small`, `medium`, `large`)
+which include [optical scaling](https://m3.material.io/styles/icons/applying-icons#b41cbc01-9b49-4a44-a525-d153d1ea1425).
```
Comment (gyfchong) [inline packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx:50]: Probably don't need to mention optical sizes here

File: packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+<Canvas of={IconStories.Sizes} />
+
+### Custom sizes
+
+Should you need a custom size, you may either `inherit` from the parent,
+or override the `font-size` (eg. `className="text-heading-1"`).
```
Comment (gyfchong) [inline packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx:57]: CSS vars to override?

File: packages/components/rollup.config.mjs (hunk 1 above)
Comment (HeartSquared) [reply packages/components/rollup.config.mjs:12]: We'll do it in a separate PR

File: packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx (hunk 4 above)
Comment (gyfchong) [reply packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx:57]: Oh, not easy to do in TW

File: packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+_Options: `20`, `24`, `40`, `48`_
+
+>For the image to look the same at different sizes, the stroke weight (thickness) changes as the icon size scales. Optical Size offers a way to automatically adjust the stroke weight when you increase or decrease the symbol size.
+
+To change the [optical size](https://m3.material.io/styles/icons/applying-icons#b41cbc01-9b49-4a44-a525-d153d1ea1425),
+override the CSS variable `--icon-optical-size` (eg. in Tailwind - `[--icon-optical-size:48]`).
+
+<Canvas of={IconStories.OpticalSize} />
+<Controls of={IconStories.OpticalSize} />
```
Comment (gyfchong) [inline packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx:69]: Should go below the accessibility section

File: packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+### Filled
+
+<Canvas of={IconStories.Filled} />
+
+### Mirror in RTL
+
+The `shouldMirrorInRTL` prop mirrors the Icon in `dir="rtl"`.
+
+Use this for directional icons based on your usage context.
```
Comment (gyfchong) [inline packages/components/src/__illustrations__/Icon/v3/_docs/Icon--api-specification.mdx:49]: Maybe explain more around the scenario for using this prop

**Comment by gyfchong** [comment]
👍

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx [hunk 7]
```diff
@@ -364,13 +365,16 @@ describe("Has validation status", () => {
...
+        description: "warning Only four waffles remain",
       })
     ).toBeInTheDocument()
   })
+
   it("announces the validation message before the Toggle's description", () => {
-    const description = "Choose you breakfast."
-    const validationMessage = "Only four waffles remain."
+    const description = "Choose your breakfast."
+    // RTL bug: Icon should be showing aria-label "caution message" instead
```
Comment (dougmacknz) [inline packages/components/src/MultiSelect/MultiSelect.spec.tsx:376]:
```suggestion
    // React Testing Library bug: Icon should be showing aria-label "caution message" instead
```

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx [hunk 8]
```diff
@@ -364,13 +365,16 @@ describe("Has validation status", () => {
     expect(
       screen.getByRole("button", {
         name: "Breakfast menu",
-        description: "Only four waffles remain",
+        // RTL bug: Icon should be showing aria-label "caution message" instead
```
Comment (dougmacknz) [inline packages/components/src/MultiSelect/MultiSelect.spec.tsx:368]:
```suggestion
        // React Testing Library bug: Icon should be showing aria-label "caution message" instead
```

File: packages/components/src/__future__/Icon/Icon.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import React from "react"
+import { render, screen } from "@testing-library/react"
```
Comment (dougmacknz) [inline packages/components/src/__future__/Icon/Icon.spec.tsx:2]: Aren't we doing these in Storybook these days?

**Review by dougmacknz — COMMENTED** [review]
Looking good! 🔥

File: packages/components/src/__future__/Icon/Icon.spec.tsx (hunk 9 above)
Comment (HeartSquared) [reply packages/components/src/__future__/Icon/Icon.spec.tsx:2]: Not for DOM-only tests :)

File: docs/components/DosAndDonts/DosAndDonts.tsx [hunk 10]
```diff
@@ -30,20 +30,16 @@ export const DoOrDont = ({
...
       )}
     >
-      {isDont ? (
-        <CloseIcon role="presentation" />
-      ) : (
-        <CheckIcon role="presentation" />
-      )}
-      <SbContent>
-        <span className="text-white">
+      <Unstyled style={{ display: "contents" }}>
```
[the remark this answers is not in the document]
Comment (HeartSquared) [reply docs/components/DosAndDonts/DosAndDonts.tsx:37]: We actually use it here as intended - Unstyled stops SB global styles leaking through, but I don't want the actual div to exist because it affects the layout.
I'll experiment later to see if I can move the Unstyled elsewhere to not need to do this.

**Review by dougmacknz — APPROVED** [review]

**Review by gyfchong — APPROVED** [review]
TL;DR (Too Long; Didn't Review)
